src/app/theme/theme.component.ts

```diff
@@ -72,12 +77,62 @@ export class ThemeComponent implements OnInit {
     this.themeService.titleColor = '#1a0dab';
     this.themeService.linkColor = '#006621';
     this.themeService.descriptionColor = '#800000';
+    revertColors();
   }
 
   terminalTheme() {
     this.themeService.backgroundColor = '#000000';
     this.themeService.titleColor = '#00ff00';
     this.themeService.linkColor = '#FFFFFF  ';
     this.themeService.descriptionColor = '#F1C40F';
+    revertColors();
+  }
+
+  nightTheme() {
+    this.themeService.backgroundColor = '#222222';
+    this.themeService.titleColor = '#5a9e26';
+    this.themeService.linkColor = '#42a2f4';
+    this.themeService.descriptionColor = '#dddddd';
+    setNight();
   }
 }
+
+function setNight() {
+    (document.getElementsByClassName("infobox")[0] as HTMLElement).style.color = '#ffffff';
```

Comment:
Hi @ValentinBercaru 

Querying DOM will slow down the page and decrease the preformance.
You should cache these variables on the top so that you query them only once.
```
var containerFluid = document.getElementsByClassName("container-fluid")[0]
```
use containerFluid in the whole codebase now, rather than querying it again.
Let me know if you have any questions.

Reply:
Hello @aayusharora and thanks for the feedback. I realized that modifying elements in that way doesn't look so good. I was thinking about adding more themeService elements in ThemeService class from `src/app/services/theme.service.ts` and then integrating them in all HTML elements from the pages which contains modified elements (and also importing `src/app/services/theme.service.ts` and including class `ThemeService` in constructor part where is needed). I think it must be as efficient as the current format of styling themes because the same method would be used for changing the elements. What do you think about this?